src/App.tsx

```diff
@@ -1,26 +1,222 @@
+
+      return;
+    }
+
+    try {
+      setTempTodo({
+        id: 0, // create a todo with id: 0
+        title: isTitle.trim(),
+        completed: false,
+        userId: USER_ID,
+      });
+
+      // викликаємо функцію createTodos, яка відправляє запит на сервер та створює новий todo
+      const newTodo = await createTodos({
+        title: isTitle.trim(),
+        completed: false,
+        userId: USER_ID,
+      });
+
+      //  додає новий елемент до масиву todos та оновлює його стан
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+      setTitle('');
+    } catch (errors) {
+      setError('Unable to add a todo');
+    } finally {
+      setTempTodo(null); // Сховати tempTodo
+    }
+  };
+
+  const deleteTodos = async (userId: number) => {
+
+    try {
+      setIsLoadingTodo(prevLoading => [...prevLoading, userId]); // Встановлення статусу завантаження для вибраного todo
+
+      await deleteTodo(userId);
+      setTodos(currentTodo => currentTodo.filter(todo => todo.id !== userId));
+    } catch (errors) {
+      setError('Unable to delete a todo');
+    } finally {
+      setIsLoadingTodo(prevLoading => prevLoading.filter(id => id !== userId)); // Видалення статусу завантаження після виконання операції
+    }
+  };
+
+  // Функція відповідає за зміну статусу завдання
+  const toggleTodoCompletion = (todoId: number) => {
+    try {
+      // Отримуємо посилання на завдання за його id
+      const updatedTodos = todos.map(todo =>
+        todo.id === todoId ? { ...todo, completed: !todo.completed } : todo,
+      );
+
+      // Оновлюємо стан todos
+      setTodos(updatedTodos);
+      setError(null);
+
+      // Отримуємо новий стан фільтра
+      let newFilterStatus = 'all';
+
+      if (filterStatus === 'completed') {
+        newFilterStatus = 'active';
+      } else if (filterStatus === 'active') {
+        newFilterStatus = 'all';
+      }
+
+      // Оновлюємо стан фільтра
+      setFilterStatus(newFilterStatus);
+    } catch (errors) {
+      setError('Unable to toggle todo completion');
+    }
+  };
+
+  // видаляє всі завершені todo
+  const clearCompletedTodos = async () => {
+    try {
+      // setLoading(true);
+      // Відбір завершених todo
+      const completedTodosIds = todos
+        .filter(todo => todo.completed)
+        .map(todo => todo.id);
+
+      // Видалення кожної завершеної todo за її ідентифікатором
+      await Promise.all(completedTodosIds.map(id => deleteTodo(id)));
+      // Оновлення списку todos, виключаючи завершені todo
+      setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+
+      // Перевірка, чи залишилися невиконані todo
+      if (todos.some(todo => !todo.completed)) {
+        setFilterStatus('completed'); // встановлюємо статус фільтра на 'Completed'todo
+      }
+    } finally {
+      // setLoading(false);
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleInputTodo = (value: string) => {
```

Comment:
`handleInputTodo` function wrapper is redundant - instead you can use  `setTitle` everywhere